CRM/Activity/BAO/Activity.php

```diff
@@ -317,6 +317,8 @@ public static function create(&$params) {
 
     // CRM-9137
     if (!empty($params['id'])) {
+      $activity = CRM_Activity_DAO_Activity::findById($params['id']);
+
       CRM_Utils_Hook::pre('edit', 'Activity', $activity->id, $params);
```

Comment:
As activity object is [already created above](https://github.com/civicrm/civicrm-core/blob/master/CRM/Activity/BAO/Activity.php#L285) and all necessary values are [copied from params](https://github.com/civicrm/civicrm-core/blob/master/CRM/Activity/BAO/Activity.php#L326) after calling pre(), i think post should be already having the necessary DAO object.

I just mention because this is the flow we follow for other components - you can check the call to pre/post in BAO/Contribution or BAO/membership for reference.